File: LeetCode/[M] - 790. Domino and Tromino Tiling/solution.py
class Solution:
    def numTilings(self, n: int) -> int:
        FF, F, T, B = 1, 1, 0, 0
        for i in range(2, n + 1):
            FF, F, T, B = F, F + FF + T + B, FF + B, FF + T
        return F % (10**9 + 7)

File: LeetCode/[M] - 790. Domino and Tromino Tiling/solution.go
package main

func numTilings(n int) int {
	FF, F, T, B := 1, 1, 0, 0
	for i := 2; i <= n; i++ {
		FF, F, T, B = F, (FF+F+T+B)%1000000007, (FF+B)%1000000007, (FF+T)%1000000007
	}
	return F % 1000000007
}
